streamdatasetjson/dataset.py

```diff
@@ -56,29 +55,35 @@ def __init__(self, dataset_json_file: JSONFileObject, name: str, prefix: str):
                 item_key = None
 
     def _raw_to_item(self, raw_item: dict) -> Item:
-        return Item(oid=raw_item["OID"],
-                    name=raw_item["name"],
-                    label=raw_item["label"],
-                    type=raw_item["type"],
-                    length=raw_item.get("length", None))
-
-    def getUniqueValues(self, variable_names: list[str], rows_to_scan: int = 0) -> dict[str, list[str]]:
-        unique={}
-
-        for colname in self._items:
-            unique[colname]=set([])
-
-        for record in self._records:
-            if rows_to_scan != 0 and scanned_rows>=rows_to_scan:
+        return Item(
+            oid=raw_item["OID"],
+            name=raw_item["name"],
+            label=raw_item["label"],
+            type=raw_item["type"],
+            length=raw_item.get("length", None),
+        )
+
+    def get_unique_values(
+        self, variables: "list[str]", rows_to_scan: int = 0
+    ) -> "dict[str, set]":
+        uniques = {name: set() for name in variables}
+
+        scanned_rows = 0
+        for row in self.observations:
+            if rows_to_scan != 0 and scanned_rows >= rows_to_scan:
                 break
-            scanned_rows=scanned_rows+1
-            for variable,value in zip(self._items,record):
-                unique[variable].add(value)
 
-        for columns in self._items:
-            unique[colname]=list(unique[colname])
+            target_item_values = [
+                (item.name, value)
+                for item, value in zip(self.items, row)
+                if item.name in variables
```

Comment:
I'm wondering if it makes sense to have this comparison case-insensitive. On one side, SAS is case-insensitive, on the other hand Dataset-JSON is not about SAS and formally all names are case-sensitive.